docs/frontend/dynamic-imports.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Dynamic Imports
+---
+
+# Dynamic Imports
+
+[Dynamic imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/#Dynamic_Imports)
+are supported in all major browsers except for Edge
+([EdgeHTML](https://en.wikipedia.org/wiki/EdgeHTML) version) and Internet
+Explorer which are both unsupported browsers for DEV. They allow you to import
+JavaScript dynamically instead of statically. Why is this important?
+Performance.
+
+Should I use them everywhere? No. They are a great tool when you need to load a
+JavaScript module on the fly for functionality that is not needed immediately
+for the page to be usable.
+
+Here are a couple of examples of this on DEV:
+
+- The
+  [Onboarding flow](https://github.com/thepracticaldev/dev.to/blob/master/app/javascript/packs/Onboarding.jsx#L28).
+- In
+  [connect](https://github.com/thepracticaldev/dev.to/blob/master/app/javascript/chat/codeEditor.jsx#L11)
+  (chat)
```

Comment:
You might want to use the permalinks here, instead of the the link, though maybe it's better to use the link because if we change those lines, then maybe someone will notice the link in the doc is broken instead of viewing the same file in a point in time, nevermind then :D